### PR TITLE
More helpful help

### PR DESCRIPTION
**Pull request by kiwiroy**
A cosmetic change to the dynamic help text.

This labels `n_of` options with a `+` and includes a notes section to explain what the `*` and `+` indicate. An example, taken from the tests, follows.

    Usage:
       --foo-bar  Foo can something
       --foo-2    foo_2 can something else
     * --foo-3    foo_3 can also something
     + --foo-4    foo_4 can also something
    
       --help     Print this help text
    
    notes:
     * denotes a required option
     + denotes an option that accepts multiple values

### CODE REVIEW

File: lib/Applify.pm [hunk 1]
```diff
@@ -170,12 +170,15 @@ OPTION:
...
     printf(
       " %s %2s%-${width}s  %s\n",
-      $option->{required} ? '*'  : ' ',
+      $option->{required} ? '*'  : $option->{n_of} ? '+' : ' ',
       length($name) > 1   ? '--' : '-',
       $name, $option->{documentation},
     );
   }
 
+  print "notes:\n";
```
Comment (jhthorsen) [inline lib/Applify.pm]: Please change this to “Notes”.

Comment (kiwiroy) [reply lib/Applify.pm]: done

File: t/help.t [hunk 2]
```diff
@@ -51,9 +52,13 @@ Usage:
    --foo-bar  Foo can something
    --foo-2    foo_2 can something else
  * --foo-3    foo_3 can also something
+ + --foo-4    foo_4 can also something
 
    --help     Print this help text
 
+notes:
```
Comment (jhthorsen) [inline t/help.t]: Did you forget to update the tests?

Comment (kiwiroy) [reply t/help.t]: Yeah, nah. Sorted now.

**Comment by jhthorsen** [comment]
Thanks! 👍

**Pull request merged**